### PR TITLE
CB-10960 Uncaught #<FileError> in write() when readyState != WRITING ?

### PR DESCRIPTION
**Pull request by daserge**
[Jira issue](https://issues.apache.org/jira/browse/CB-10960)

### CODE REVIEW

**Comment by nikhilkh** [comment]
@rakatyal  @jasongin Can you please review?

**Comment by daserge** [comment]
Please find the issue details in the [Jira](https://issues.apache.org/jira/browse/CB-10960).

File: www/FileWriter.js [hunk 1]
```diff
@@ -107,8 +108,29 @@ FileWriter.prototype.write = function(data) {
         var fileReader = new FileReader();
         fileReader.onload = function() {
             // Call this method again, with the arraybuffer as argument
-            FileWriter.prototype.write.call(that, this.result);
+            FileWriter.prototype.write.call(that, this.result, true /* isPendingBlobReadResult */);
         };
+        fileReader.onerror = function (evt) {
+            // DONE state
+            that.readyState = FileWriter.DONE;
```
Comment (jasongin) [inline www/FileWriter.js]: Do you need to check if this FileWriter is already in the DONE (cancelled) state, as in the other callbacks below?

File: www/FileWriter.js [hunk 2]
```diff
@@ -107,8 +108,29 @@ FileWriter.prototype.write = function(data) {
...
+
+            // Save error
+            that.error = evt.target.error;
```
Comment (jasongin) [inline www/FileWriter.js]: I think **evt.target** here is the same as **this** in the callback, and they both refer to the FileReader instance, right? It's a bit confusing that you use **evt.target** here, but **this** is used in the **onload** callback. (I can't actually find documentation that says **this** in a FileReader callback is set to the FileReader, though it seems like a reasonable assumption.)

**Comment by jasongin** [comment]
LGTM aside from the minor issues I commented on.

File: www/FileWriter.js (hunk 2 above)
Comment (rakatyal) [inline www/FileWriter.js]: +1

**Comment by rakatyal** [comment]
LGTM as well aside from Jason's feedback.

File: www/FileWriter.js (hunk 1 above)
Comment (daserge) [inline www/FileWriter.js]: I don't think so as the `DONE` state is being set in `FileReader.readSuccessCallback`, `FileReader.readFailureCallback` and `FileReader.abort` only.
`abort` should not be called outside (and we don't call it) taking into account that fileReader is an internal variable, `readSuccessCallback` calls `onload` and `readFailureCallback` calls `onerror` so we are covering all the cases.

File: www/FileWriter.js (hunk 2 above)
Comment (daserge) [inline www/FileWriter.js]: Thanks, I will update it!

**Comment by daserge** [comment]
Thank you for the review!

**Pull request merged**